analyses/cell-type-ewings/exploratory_analysis/04-infercnv.Rmd

```diff
@@ -0,0 +1,480 @@
+  dplyr::mutate(
+    has_cnv = ifelse(
+      stringr::str_detect(subcluster, "reference"), 
+      NA_real_, 
+      rowSums(across(has_cnv_cols))
+    ),
+    z_scaled_cnv = scale(has_cnv))
+```
+
+
+```{r}
+# plot the total number of CNVs
+ggplot(cnv_df, aes(x = UMAP1, y= UMAP2, color = has_cnv)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  theme_bw() +
+  scale_color_viridis_c()
+```
+This shows that the total CNVs are highest in the group of cells that correspond to the same cells that are manually annotated as tumor cells, which is promising. 
+Those cells in grey (bottom left) are cells used as a reference. 
+If we were to use total CNVs we would need some sort of cut off for the number. 
+Let's look at the distribution of total CNVs and also the scaled total CNVs to see if we can identify a cut off. 
+
+```{r}
+ggplot(cnv_df, aes(x = has_cnv)) +
+  geom_density() +
+  theme_bw() +
+  labs(x = "Total number of CNVs")
+
+ggplot(cnv_df, aes(x= z_scaled_cnv)) + 
+  geom_density() +
+  theme_bw() +
+  labs(x = "z-scaled chromosome CNV count")
+```
+
+Looking at the scaled number of CNVs it does look like we could use 0 as the cutoff and call any cell that has > 0 CNVs (after scaling) as tumor and all other cells as normal. 
+
+In the below plot we label all non-reference cells as either tumor or normal. 
+Cells that are NA are those that were already classified as reference cells. 
+
+```{r}
+# add cnv classification 
+cnv_df <- cnv_df |> 
+  dplyr::mutate(cnv_classification = dplyr::if_else(z_scaled_cnv > 0, "Tumor", "Normal"))
+
+ggplot(cnv_df, aes(x = UMAP1, y= UMAP2, color = cnv_classification)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  theme_bw() 
+```
+
+Another option is to use the scaled proportion of CNVs which tells us the proportion of genes on that chromosome that are part of the CNV.  
+
+```{r}
+proportion_cols <- names(cnv_df)[startsWith(colnames(cnv_df), "proportion_scaled_cnv")]
+
+# get total number of genes per chromosome to use in weighted mean 
+chr_weights <- final_cnv_obj@gene_order |> 
+  as.data.frame() |> 
+  dplyr::count(chr) |> 
+  # only keep chr 1-22, drops MT and GL chr
+  dplyr::filter(chr %in% glue::glue("chr{seq(1,22)}")) |> 
+  dplyr::pull(n)
+
+# get mean proportion of cnvs per cell weighted by chromsome size (number of genes)
+cnv_df <- cnv_df |> 
+  dplyr::rowwise() |> 
+  # calculate weighted mean, excluding normal cells 
+  dplyr::mutate(
+    mean_proportion = ifelse(
+      !is.na(has_cnv), 
+      weighted.mean(across(proportion_cols), chr_weights), 
+      NA)) |> 
+  dplyr::ungroup()
+```
+
+```{r}
+ggplot(cnv_df, aes(x = UMAP1, y= UMAP2, color = mean_proportion)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  theme_bw() +
+  scale_color_viridis_c()
+```
+
+```{r}
+ggplot(cnv_df, aes(x = mean_proportion)) +
+  geom_density() +
+  theme_bw()
+```
+
+It looks like we could use a cutoff of 0.1 here to classify tumor/normal cells. 
```

Comment:
I might also look at using the mean here? This will be a more productive discussion with more samples, of course.

Reply:
We are taking the weighted mean here so want to check that you just want to use the mean alone? Or compare calling tumor cells using the weighted mean to the mean of the CNV sum?

Reply:
I think yes. Specifically, the cutoff I am proposing is `mean(cnv_df$mean_proportion)`

Reply:
Sorry I was confused and thought you meant to look at the distribution of the mean and wasn't clear that you meant to use it as the cutoff. But that makes sense so I'm updating it to use that!